typing/typemod.ml

```diff
@@ -673,6 +754,25 @@ let simplify_signature sg =
   let (sg, _) = aux sg in
   sg
 
+
+let remove_inserted_modtype mty =
+  let remove =
+    List.filter (
+      function
+      | Sig_module({Ident.name}, _, _) when String.contains name '#' -> false
```

Comment:
This feels a bit brittle and not really informative, would it be possible to define a `is_open_generated` (or a better name) function rather than use `String.contains name '#' `?

Reply:
I'd also suggest inlining remove where it is used (in the case for Mty_signature), and drop the `aux` inner function (just making `remove_inserted_modtype` recursive).  (Btw, it's not yet obvious to me why we need to recurse under mty_arg -- I haven't reviewed all the PR yet.)

Reply:
@alainfrisch 

One can `open`in functor argument signature

`module M (S: sig open struct ... end ... end) = struct ... end`